plugin/src/main/java/com/redhat/jenkins/nodesharingfrontend/Api.java

```diff
@@ -219,14 +229,10 @@ public Response doPostRequest(@Nonnull final WebTarget target, @Nonnull final St
      */
     @CheckForNull
     public Object nodeStatus(@Nonnull @QueryParameter("name") final String name) {
-        // TODO Response to node status
-        Communication.NodeState status = Communication.NodeState.NOT_FOUND;
-        Node node = Jenkins.getActiveInstance().getNode(name);
-        if (node != null) {
-            // TODO Extract the current states
-            status = Communication.NodeState.FOUND;
+        if(name == null) {
+            throw new ActionFailed.CommunicationError("Node name cannot be 'null'!");
```

Comment:
This is an API abuse - `IllegalArgumentException`.

Reply:
:+1: